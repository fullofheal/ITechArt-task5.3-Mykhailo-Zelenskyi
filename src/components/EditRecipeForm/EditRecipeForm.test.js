import { cleanup, screen, render } from "@testing-library/react";
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../store';
import EditRecipeForm from './EditRecipeForm';

afterEach(() => {
  cleanup();
});

const blankRecipe = {
  id: '',
  country: '',
  category: '',
  instructions: '',
  name: '',
  tags: '',
  ingredients: [['','']],
  picture: '',
  video: ''
};

describe('test if component is rendered on the page', () => {

  test('should render EditRecipeForm component', () => {
    const details = {
      ...blankRecipe
    };
    render(
      <Provider store={store}>
        <Router>
          <EditRecipeForm recipeDetails={details} />
        </Router>
      </Provider>
    );
    const EditRecipeFormComponent = screen.getByTestId('EditRecipeForm');
    expect(EditRecipeFormComponent).toBeInTheDocument();
  });
});

test('should render RecipeVideo component inside EditRecipeForm component when url is passed', () => {
  const details = {
    ...blankRecipe,
    video: 'https://www.youtube.com/watch?v=kYH2qJXnSMo'
  };
  render(
    <Provider store={store}>
      <Router>
        <EditRecipeForm recipeDetails={details}/>
      </Router>
    </Provider>
  );
  const iFrame = screen.getByTestId('RecipeVideo');
  expect(iFrame).toBeInTheDocument();
});

test('EditRecipeForm matches snapshot', () => {
  const details = {
    ...blankRecipe,
  };
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <EditRecipeForm recipeDetails={details}/>
      </Router>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});