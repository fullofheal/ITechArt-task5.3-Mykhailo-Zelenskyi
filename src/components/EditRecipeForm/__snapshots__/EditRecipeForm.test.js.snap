// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRecipeForm matches snapshot 1`] = `
<form
  className="form"
  data-testid="EditRecipeForm"
  onSubmit={[Function]}
>
  <div
    className="form__picture"
  >
    <img
      alt=""
      src=""
    />
    <label
      className="form__label"
      htmlFor="picture-url"
    >
      Picture url
    </label>
    <input
      name="picture-url"
      onChange={[Function]}
      placeholder="Provide valid image url"
      type="text"
      value=""
    />
  </div>
  <div>
    <label
      className="form__label"
      htmlFor="name"
    >
      Name
    </label>
    <input
      name="name"
      onChange={[Function]}
      pattern="[^#$^&*?><\\\\/0-9]+"
      placeholder="Recipe name"
      required={true}
      type="text"
      value=""
    />
    <span>
      Only letters
    </span>
  </div>
  <div>
    <label
      className="form__label"
      htmlFor="tags"
    >
      Tags
    </label>
    <input
      name="tags"
      onChange={[Function]}
      pattern="[A-Za-z, ]+"
      placeholder="List associated tags"
      type="text"
      value=""
    />
    <span>
      Only letters
    </span>
  </div>
  <div>
    <label
      className="form__label"
      htmlFor="country"
    >
      Country of origin
    </label>
    <input
      name="country"
      onChange={[Function]}
      pattern="[A-Za-z ]+"
      placeholder="List associated tags"
      type="text"
      value=""
    />
    <span>
      Only letters
    </span>
  </div>
  <label
    className="form__label"
    htmlFor="instructions"
  >
    Instructions
  </label>
  <textarea
    className="form__instructions"
    name="instructions"
    onChange={[Function]}
    placeholder="Provide detailed preparation instructions"
    required={true}
    value=""
  />
  <div
    className="add-form__ingredients"
  >
    <div
      className="form__ingredient"
    >
      <div>
        <input
          name=""
          onChange={[Function]}
          pattern="[^#$^&*?><\\\\/0-9]+"
          placeholder="enter ingredient name"
          required={true}
          type="text"
          value=""
        />
        <span>
          Only letters
        </span>
      </div>
      <div>
        <input
          name=""
          onChange={[Function]}
          pattern="[^#$^&*?><]+"
          placeholder="enter the measure"
          type="text"
          value=""
        />
        <span>
          Invalid symbols used
        </span>
      </div>
      <button
        onClick={[Function]}
        type="button"
      >
        Remove
      </button>
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      Add new ingredient
    </button>
  </div>
  <label
    className="form__label-video"
    htmlFor="video-url"
  >
    Video url
  </label>
  <input
    name="video-url"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Submit recipe
  </button>
</form>
`;
